crates/spartan-farmer/src/commands/dht/client.rs

```diff
@@ -0,0 +1,308 @@
+use super::*;
+use super::{core::create_node, eventloop::EventLoop};
+use libp2p::kad::QueryInfo;
+
+#[derive(Copy, Clone, Debug)]
+pub enum ClientType {
+    // Bootstrap node. It uses the following fields from `ClientConfig`:
+    // 1. `bootstrap_keys`: Private keys/private key location to create bootstrap node peerId.
+    // 2. `listen_addr`: Listening address for Bootstrap node.
+    Bootstrap,
+    // Normal node. It uses the following fields from `ClientConfig`:
+    // 1. `bootstrap_nodes`: Bootstrap nodes addresses that the normal node must connect to.
+    // For setting listening address, use client.start_listening.
+    Normal,
+}
```

Comment:
2 notes here:
1) Bootstrap-only nodes will probably be a separate from farmer applications
2) Every farmer should be able to act as a bootstrap node, so I don't think we need this distinction in the farmer: every farmer should have an ability to join other bootstrap nodes and act as bootstrap node for others

The primary reason is that when we have a healthy network, we don't want farmers to reconnect to the same centralized bootstrap nodes on restart, they should be able to join other bootstrap nodes they have discovered before.

Reply:
In our current implementation, normal nodes can (and, are) acting as bootstrap nodes. If you take a look at the test, you will notice that all the nodes are created as normal nodes. The `ClientType` enum, does not do anything useful, at this point.

Reply:
Let's do it then. The easier things to begin with, the better